GOVERNANCE.md

```diff
@@ -0,0 +1,53 @@
+# CoreDNS Governance
+
+## Principles
+
+The CoreDNS community adheres to the following principles:
+
+- Open: CoreDNS is open source. See repository guidelines.
+- Welcoming and respectful: See Code of Conduct, below.
+- Transparent and accessible: Work and collaboration are done in public.
+- Merit: Ideas and contributions are accepted according to their technical merit and alignment with project objectives, scope, and design principles.
+
+## Voting
+
+The CoreDNS project employs "organization voting" to ensure no single organization can dominate the project.
+
+Individuals not associated with or employed by a company or organization are allowed one organization vote. Each company or organization (regardless of the number of maintainers associated with or employed by that company/organization) receives one organization vote.
+
+In other words, if two maintainers are employed by Company X, two by Company Y, two by Company Z, and one maintainer is an un-affiliated individual, a total of four "organization votes" are possible; one for X, one for Y, one for Z, and one for the un-affiliated individual.
+
+Any maintainer from an organization may cast the vote for that organization.
+
+For formal votes, a specific statement of what is being voted on should be added to the relevant github issue or PR, and a link to that issue or PR added to the maintainers meeting agenda document. Maintainers should indicate their yes/no vote on that issue or PR, and after a suitable period of time, the votes will be tallied and the outcome noted.
+
+## Changes in Maintainership
+
+New maintainers are proposed by an existing maintainer and are elected by a 2/3 majority organization vote.
+
+Maintainers can be removed by a 2/3 majority organization vote.
```

Comment:
Also especially personal contributors might go into read-only mode for a bit due to other work taking priority, but come back more actively afterwards. Focusing on the full-time, corporate backed activity might result in personal contributors being considered inactive and push them out. But maybe that's a special case, but this always bugged me in other projects such as K8s.

Reply:
> Focusing on the full-time, corporate backed activity might result in personal contributors being considered inactive and push them out.

It completely depends on how we define "active".... and how long in-activity is permitted.   If the required activity level is very high, then corporate backed contributors get more voting power.  If a very low level of activity is permitted, with long periods of complete in-activity, then individuals get more collective voting power.  Is Google limited to one vote in K8s?  

As Yong says, I think we'll want to define more precisely what "contributing" means... but I assume it broadly means commits, code reviews, and or participation in meetings and issue discussions, relating to any repositories in the CoreDNS project. Also, we'll want to define required frequency of contributions to be considered active.

Reply:
@yongtang : Looks like I misunderstood you comment above.
You request a change : can you be more specific on what text you would like to add or change or delete ?

>Software have life cycles and maintainers on any projects have as well. In the past Hadoop, OpenStack may have those as well. Though in order to avoid drama (saw that happen in some other open source communities), it might be better to specify what will trigger a removal. Ethnical or no contribution? May want to specify at least some criteria.

Reply:
I think CoreDNS maintainers serve the bridges between community members. Reviewing PRs (with Code of Conduct) is the best bridge for the community.
```
## Expectations of Maintainers

Maintainers are the bridges between members of the CoreDNS community. Maintainers
actively participate in PR reviews. Maintainers are expected to respond to
assigned PRs in a reasonable time frame, either providing insights, or assign the
PRs to other maintainers.
```

```
## Changes in Maintainership

Maintainers who fail to meet the principles of CoreDNS community can be removed
by a 2/3 majority organization vote.
```